mono/utils/mono-threads.c

```diff
@@ -1144,6 +1144,12 @@ mono_thread_info_abort_socket_syscall_for_close (MonoNativeThreadId tid)
 	}
 
 	mono_thread_info_suspend_lock ();
+	/* lookup info again in case we raced with shutdown of the other thread and it's already gone.*/
+	info = mono_thread_info_lookup (tid);
```

Comment:
The duplication doesn't seem warranted, so we should just remove https://github.com/lambdageek/mono/blob/875a9deb369778b32409109b21c37d45b9f7f06d/mono/utils/mono-threads.c?utf8=%E2%9C%93#L1137-L1144 (we shouldn't need to keep the check for `STATE_DETACHED` since the removing of the `MonoThreadInfo` on detach is done while holding the `suspend_lock`.